### PR TITLE
PBL-7122 Add code collapse feature in editor

### PR DESCRIPTION
**Pull request by Spacerat**
The extension all seems to work fine for C and JS. The folds reset on page reload, but we could eventually make them persist (at some point).

(Also, I'm not 100% sure on whether ide.css is the right place to put the extra CSS.)

### CODE REVIEW

**Comment by Spacerat** [comment]
Closing this for now, until we decide how to finish it (i.e. storing in cloud vs local vs none)

**Comment by Spacerat** [comment]
Reopened now with an implementation that stores the code folds server-side (as a comma-separated string of the line numbers of folds, per file)

**Comment by Katharine** [comment]
I would prefer to see a couple of changes here:

* Store the folded lines as ranges, not a list of individual lines
* Store the folded lines in JSON instead of making up a format.

File: ide/api/source.py [hunk 1]
```diff
@@ -90,7 +97,7 @@ def save_source_file(request, project_id, file_id):
                 }
             }, project=project, request=request)
             raise Exception(_("Could not save: file has been modified since last save."))
-        source_file.save_file(request.POST['content'])
+        source_file.save_file(request.POST['content'], folded_lines = request.POST['folded_lines'])
```
Comment (Katharine) [inline ide/api/source.py]: No spaces around equals signs in named arguments.

**Pull request merged**